src/libraries/System.Reflection.Context/pkg/System.Reflection.Context.pkgproj

```diff
@@ -16,8 +16,10 @@
     <ValidatePackageSuppression Include="TreatAsOutOfBox">
       <Value>UAP</Value>
     </ValidatePackageSuppression>
-
-    <HarvestIncludePaths Include="ref/netcore50;lib/netcore50;ref/portable-net45+win8;lib/portable-net45+win8" />
   </ItemGroup>
+  <!-- TODO: Re-enable support check after the 6.0 version of the package is released. -->
```

Comment:
Can you elaborate?  What was going wrong here?

Reply:
These were the errors that I wasn't sure how to deal with them:

```
Errors
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETCore,Version=v4.5 should support System.Reflection.Context inbox but contained a reference assemblies: ref/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="netcore45" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETCore,Version=v4.5 should support System.Reflection.Context inbox but contained a implementation assemblies: lib/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="netcore45" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETCore,Version=v4.5.1 should support System.Reflection.Context inbox but contained a reference assemblies: ref/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="netcore451" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETCore,Version=v4.5.1 should support System.Reflection.Context inbox but contained a implementation assemblies: lib/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="netcore451" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETPortable,Version=v4.6,Profile=Profile44 should support System.Reflection.Context inbox but contained a reference assemblies: ref/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="portable46-net451+win81" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETPortable,Version=v4.6,Profile=Profile44 should support System.Reflection.Context inbox but contained a implementation assemblies: lib/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="portable46-net451+win81" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETPortable,Version=v4.5,Profile=Profile7 should support System.Reflection.Context inbox but contained a reference assemblies: ref/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="portable45-net45+win8" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework .NETPortable,Version=v4.5,Profile=Profile7 should support System.Reflection.Context inbox but contained a implementation assemblies: lib/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="portable45-net45+win8" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework Windows,Version=v8.0 should support System.Reflection.Context inbox but contained a reference assemblies: ref/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="win8" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
    C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): Framework Windows,Version=v8.0 should support System.Reflection.Context inbox but contained a implementation assemblies: lib/netstandard1.1/System.Reflection.Context.dll.  You may need to add <InboxOnTargetFramework Include="win8" /> to your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
```

I made sure that the generated nupkg has the expected state, that's why I disabled the support check.

Reply:
This is because S.R.C is marked as inbox in the package index.  The inbox validation doesn't understand that you can decide to not represent an assembly that's inbox (here it would just get a placeholder). 

I think you can make this go away by adding `<InboxOnTargetFramework Include="portable-net45+win8" />`  It looks like that Harvesting was only harvesting the placeholder files.

Reply:
I can't get this working. After adding the InboxOnTargetFramework items for both portables and win8, I now get the following errors:

```
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-x86 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-x86-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-x64 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-x64-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-arm but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on .NETCore,Version=v5.0/win10-arm-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="netcore50" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-x86 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-x86-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-x64 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-x64-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-arm but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0/win10-arm-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-x86 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-x86-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-x64 but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-x64-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-arm but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
C:\Users\vihofer\.nuget\packages\microsoft.dotnet.build.tasks.packaging\6.0.0-beta.21221.6\build\Packaging.targets(1119,5): error : System.Reflection.Context should be supported on UAP,Version=v10.0.16299/win10-arm-aot but has a compile placeholder.  You may need to remove InboxOnTargetFramework Include="uap10.0.16299" /> from your project. [C:\git\runtime2\src\libraries\System.Reflection.Context\pkg\System.Reflection.Context.pkgproj]
```

If there isn't a quick solution to it I'd be fine to keep the support check for this package disabled.

Reply:
I see, because those are OOB frameworks, so the package has to provide an asset.  One solution would be to remove the inbox entries from the index just for this package.  That would avoid the original issue you mentioned.  
https://github.com/dotnet/runtime/blob/1cc11b1fadfd508b2d906e1f2706e1835b9cf014/src/libraries/pkg/baseline/packageIndex.json#L4908-L4913

Remove the `portable` entries and `win8` (effectively "lying" about what's in those targeting packs).  Alternatively you could plumb a suppression that told validation to not check for support for some frameworks (take the existing hook you added and consider using it in more places) and consume that near here: https://github.com/dotnet/arcade/blob/e7ede87875f41a9b3df898ae08da5ebc96e24f56/src/Microsoft.DotNet.Build.Tasks.Packaging/src/ValidatePackage.cs#L585

Reply:
Thanks a lot for the suggestion Eric. I decided to trim the packageIndex graph as I wouldn't want to spend much on a solution for obsolete frameworks. Either the deprecation of the packageIndex or the removal of old netstandard1.x assets will happen at some point which will hopefully make the consideration of these old framework unnecessary.